### ISSUE TITLE
Bug: DTO partial does not work with SQLAlchemy UUIDBase

### ISSUE THREAD

**Issue by MatthewNewland** [litestar-org/litestar#1685]
### Description

For SQLAlchemy models that inherit from the provided `UUIDBase` or `UUIDAuditBase` classes, we get an exception on app startup if there is a DTO defined with `partial = True`.

I think I have a fix for it, I will probably open a PR in just a minute.

### MCVE

```python
from litestar import Litestar, patch
from litestar.contrib.sqlalchemy.base import UUIDBase
from litestar.contrib.sqlalchemy.dto import SQLAlchemyDTO
from litestar.dto.factory import DTOConfig, DTOData
from litestar.testing import TestClient
from sqlalchemy.orm import Mapped


class MyModel(UUIDBase):
    foo: Mapped[str]
    bar: Mapped[str]


class MyModelDTO(SQLAlchemyDTO[MyModel]):
    pass


class MyModelPatchDTO(SQLAlchemyDTO[MyModel]):
    config = DTOConfig(partial=True)


@patch("/", dto=MyModelPatchDTO, return_dto=MyModelDTO)
async def update_model_partial(data: DTOData[MyModel]) -> MyModel:
    model = MyModel(foo="foo", bar="bar")
    data.update_instance(model)
    return model


app = Litestar(
    debug=True,
    route_handlers=[update_model_partial],
)


if __name__ == "__main__":
    client = TestClient(app)
    client.patch("/", json={"foo": "baz"})
```

### Steps to reproduce

```bash
1. Run the code above.
2. You'll get a stack trace ending in:
 _Field(default=default, default_factory=default_factory, name=name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Cannot set both `default` and `default_factory
```

### Litestar Version

2.0.0.alpha7

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

**Comment by MatthewNewland** [comment]
See #1686

**Comment by peterschutt** [comment]
Closed in #1686
